File: part1-exercises-for-rust/hardways/leetcode-remove-element/solution.c
int removeElement(int* nums, int numsSize, int val){
    int size = 0;
    int i = 0;

    for (i = 0; i <= numsSize - 1; i++)
    {
        if (nums[i] != val) // ignore the same
        {
            nums[size++] = nums[i];
        }
    }

    return size;
}

File: part1-exercises-for-rust/hardways/leetcode-remove-element/solution.rs
impl Solution {
    pub fn remove_element(mut nums: &mut Vec<i32>, val: i32) -> i32 {
        let mut index = 0;
        let mut count = 0;
        let mut a = Vec::<i32>::new();
        for i in 0..nums.len(){
            if nums[i] != val {
                nums[index] =  nums[i].clone();
                index = index+1;
                count = count+1;
            }
        }
        count
    }
}